app/controllers/filter.ts

```diff
@@ -116,8 +118,9 @@ export default class FilterController extends Controller {
   updateDistance(selectedDistance: DistanceOption) {
     this.distanceList.selected = selectedDistance;
     this.filterService.updateFilters({
-      distance: selectedDistance,
+      distance: selectedDistance.id,
```

Comment:
This is possible to have a null value as selected
![image](https://github.com/user-attachments/assets/3ef29189-bfa7-447f-bc89-6597600c4a1d)

```suggestion
      distance: selectedDistance?.id,
```
This should do the trick I think